src/Fixer/Whitespace/HeredocIndentationFixer.php

```diff
@@ -107,19 +108,35 @@ private function fixIndentation(Tokens $tokens, $start, $end)
                 continue;
             }
 
-            $regexEnd = $last && !$currentIndent ? '(?!$)' : '';
-            $content = Preg::replace('/(?<=\v)'.$currentIndent.$regexEnd.'/', $indent, $tokens[$index]->getContent());
+            $content = $tokens[$index]->getContent();
+
+            if ($currentIndent) {
```

Comment:
Aren't we usually explicit? (Also in the ternary on line 117 and the `elseif` on line 135)

Reply:
Indeed, we prefer explicit comparisons.